environment-variables/validate-environment-variables/snippet.go

```diff
@@ -0,0 +1,28 @@
+package snippets
+
+import (
+	"github.com/kelseyhightower/envconfig"
+)
+
+// See https://godoc.org/github.com/kelseyhightower/envconfig
+type Specification struct {
+	Regular              string
+	Array                []int
+	Map                  map[string]int
+	SplitWithUnderscores int `split_words:"true"`
+	WithDefaultValue     int `default:"42"`
+	OverriddenName       int `envconfig:"MANUAL_OVERRIDE"`
+	EmbeddedStruct       EmbeddedSpecification
+	Required             int `required:"true"`
+}
+
+type EmbeddedSpecification struct {
+	Value int
```

Comment:
I don't really understand at a glance what this "embedded" thing is about.  First, it's kind of unusual to have a structured value passed in as an env var, they're almost always scalar values.  Let's use the [Pareto principle](https://en.wikipedia.org/wiki/Pareto_principle) here :p  What's the most basic use -- not aiming to be comprehensive. Otherwise it quickly gets overwhelming to even grok the basics.

Reply:
:+1: